indigo/indigo/src/main/scala/indigo/shared/events/GlobalEvent.scala

```diff
@@ -289,6 +289,18 @@ object KeyboardEvent {
   */
 final case class PlaySound(assetName: AssetName, volume: Volume) extends GlobalEvent
 
+/** A class of events representing general networking events
+  */
+trait NetworkEvent extends GlobalEvent
+
+/** The network has come online and is now available
+  */
+case object Online extends NetworkEvent
+
+/** The network has gone offline and is now unavailable
+  */
+case object Offline extends NetworkEvent
```

Comment:
Rather than looking at things like `NetworkSendEvent`, which is an oddity, can you look at things like `StorageEvent` for the convention here? Specifically it ought to be a `sealed trait`, and the `Online`/`Offline` placed inside a companion object*.

(* This pattern was considered good practice in Scala 2, and Scala 3's enums which are kind of a replacement now enforce it.)